aten/src/ATen/AccumulateType.h

```diff
@@ -104,6 +105,25 @@ MPS_ACC_TYPE(c10::complex<Half>, c10::complex<float>);
 MPS_ACC_TYPE(c10::complex<float>, c10::complex<float>);
 MPS_ACC_TYPE(c10::complex<double>, c10::complex<float>);
 
+XPU_ACC_TYPE(BFloat16, float);
+XPU_ACC_TYPE(Half, float);
+XPU_ACC_TYPE(Float8_e5m2, float);
+XPU_ACC_TYPE(Float8_e4m3fn, float);
+XPU_ACC_TYPE(Float8_e5m2fnuz, float);
+XPU_ACC_TYPE(Float8_e4m3fnuz, float);
+XPU_ACC_TYPE(float, float);
```

Comment:
Shouldn't do have double here like we do for cpu?

Reply:
Currently, XPU adopts the same mapping strategy as CUDA

Reply:
Why not follow the CPU logic though? Unless you have strict hardware limitation, the higher precision is what users expect and it will avoid a lot of issues of users complaining for loss of precision.

Reply:
Yes, partially agree. Here should be trad-off between precision and performance, I think. Single precision floating point used in accumulation/reduction is broadly acceptable both in PyTorch defined unit tests and AI/DL workload. Here, when precision is acceptable, we prefer float not double, since one more clock might be paid in some platform.

Reply:
In terms of functionality, we have two considerations. First, some GPUs don't support double precision calculations. Second, using the same acc type as CUDA on the GPU can help avoid some unit testing issues biased towards CUDA, which facilitates our current development.